routes/offerings.js

```diff
@@ -0,0 +1,126 @@
+const Posting = require("../models/Posting");
+const Notifications = require("../utils/notificationsUtil");
+
+// http://localhost/api/offerings/{offeringId}/accept ///////////////////////////////////////////////////////
+// POST
+// Accepts the indicated offering
+router.post("/:offeringId/accept", async (req, res) => {
+  try {
+    const offering = await Offering.findOneAndUpdate(
+      {
+        _id: req.params.offeringId,
+      },
+      {
+        $set: { status: "ACCEPTED" },
+      },
+      {
+        new: true,
+      }
+    );
+
+    await Notifications(
+      offering.userId,
+      "offeringAccepted",
+      "Your offering has been accepted!"
+    );
+
+    const allPostingOfferings = await Posting.findOneAndUpdate(
+      {
+        _id: offering.postingId,
+      },
+      {
+        $set: { active: false },
+      }
+    ).offerings;
+
+    allPostingOfferings.forEach(async (offeringId) => {
+      if (offeringId !== req.params.offeringId) {
+        const rejectedOffering = await Offering.findOneAndUpdate(
+          {
+            _id: offeringId,
+          },
+          {
+            $set: { status: "REJECTED" },
+          },
+          {
+            new: true,
+          }
+        );
+
+        await Notifications(
+          rejectedOffering.userId,
+          "offeringRejected",
+          "Your offering was rejected"
+        );
+      }
+    });
+
+    res.status(200).json(offering);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+// http://localhost/api/offerings/{offeringId}/rescind ///////////////////////////////////////////////////////
+// POST
+// Rescinds the indicated offering
+router.post("/:offeringId/rescind", async (req, res) => {
+  try {
+    const offering = await Offering.findOneAndUpdate(
+      {
+        _id: req.params.offeringId,
+      },
+      {
+        $set: { status: "RESCINDED" },
+      },
+      {
+        new: true,
+      }
+    );
+
+    res.status(200).json(offering);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
+
+// http://localhost/api/offerings/{offeringId}/reject ///////////////////////////////////////////////////////
+// POST
+// Rejects the indicated offering
+router.post("/:offeringId/reject", async (req, res) => {
+  try {
+    const offering = await Offering.findOneAndUpdate(
+      {
+        _id: req.params.offeringId,
+      },
+      {
+        $set: { status: "REJECTED" },
+      },
+      {
+        new: true,
+      }
+    );
+
+    await Notifications(
+      offering.userId,
+      "offeringRejected",
+      "Your offering was rejected"
+    );
+
+    res.status(200).json(offering);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: "Error code 500: Failed to process request",
+    });
+  }
+});
```

Comment:
Missing export of this router and declaration of this route in `App.js` file

Reply:
Fixed